src/system/sargon_os/sargon_os_security_structures.rs

```diff
@@ -0,0 +1,306 @@
+use crate::prelude::*;
+
+#[uniffi::export]
+impl SargonOS {
+    /// Returns all the SecurityStructuresOfFactorSources,
+    /// by trying to map FactorSourceID level -> FactorSource Level
+    pub fn security_structures_of_factor_sources(
+        &self,
+    ) -> Result<SecurityStructuresOfFactorSources> {
+        self.profile_holder
+            .access_profile_with(|p| p.security_structures_of_factor_sources())
+    }
+
+    /// Returns all the `SecurityStructuresOfFactorSourceIDs` which are stored
+    /// in profile.
+    pub fn security_structures_of_factor_source_ids(
+        &self,
+    ) -> SecurityStructuresOfFactorSourceIDs {
+        self.profile_holder.access_profile_with(|p| {
+            p.app_preferences
+                .security
+                .security_structures_of_factor_source_ids
+                .clone()
+        })
+    }
+
+    /// Returns all the `SecurityStructuresOfFactorSourceIDs` which are stored
+    /// in profile.
+    pub fn security_structure_of_factor_sources_from_security_structure_of_factor_source_ids(
+        &self,
+        structure_of_ids: &SecurityStructureOfFactorSourceIDs,
+    ) -> Result<SecurityStructureOfFactorSources> {
+        self.profile_holder.try_access_profile_with(|p| {
```

Comment:
When should we use `access_profile_with` and when the `try_access_profile_with` ?

Reply:
Should be easy to know, here I'm calling a faillable function, `SecurityStructureOfFactorSources::try_from`, so I **MUST** use `try_access_profile_with`. 

And we really SHOULD NOT use faillable unnecessarily, so when in doubt use a non-faillable one, and then the Rust compiler will yell at you if you call something faillable :D

Reply:
I don't know what was I thinking last week, it is perfectly clear to me now